Docs & Standards: Improve PHPDoc consistency and grammar in functions.php

prab18hat opened the pull request:
- Updated @return type for is_new_day() from int to bool
- Clarified return type in xmlrpc_getpostcategory()
- Improved @param description for get_weekstartend()
- Corrected grammar in build_query() docblock
- Improved return description in get_status_header_desc()

<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

Summary of Changes:

Standardized PHPDoc for is_new_day():

Updated the @return type from int to bool for clarity and modern PHPDoc standards.

Grammar and Docblock Improvements in functions.php:

xmlrpc_getpostcategory(): Clarified the return type and simplified awkward phrasing for readability.

get_weekstartend(): Improved the @param description for $start_of_week to clearly state “Default is an empty string.”

build_query(): Corrected grammar by adding the missing article and removing an unnecessary comma.

Old: Builds URL query based on an associative and, or indexed array.

New: Builds a URL query based on an associative or indexed array.

get_status_header_desc(): Updated return description to: “Status description if found, or an empty string otherwise.” for better clarity.




Trac ticket: Trac ticket: https://core.trac.wordpress.org/ticket/63892


---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

prab18hat committed:
Docs: Standardize PHPDoc and fix grammar in functions.php

- Updated @return type for is_new_day() from int to bool
- Clarified return type in xmlrpc_getpostcategory()
- Improved @param description for get_weekstartend()
- Corrected grammar in build_query() docblock
- Improved return description in get_status_header_desc()

dhruvang21 commented:
@prab18hat you also need to update the test cases code to pass the phpunit tests 

```
public function data_is_new_date() {
		return array(
			array( '21.05.19', '21.05.19', false ),
			array( '21.05.19', '20.05.19', true ),
			array( '21.05.19', false, true ),
		);
	}
```

prab18hat committed:
Tests: Update is_new_day() tests to expect boolean return values

prab18hat commented:
Thanks for the feedback! I’ve updated the test cases for is_new_day() so that the data provider now returns booleans (true/false). The PHPUnit assertions now match the updated return type.

swissspidy commented:
🚨  We cannot just change the return type of `is_new_day()`. That would be a breaking change. We need to ensure backward compatibility. Please revert those changes.

dhruvang21 commented on tests/phpunit/tests/functions/isNewDay.php:
```diff
@@ -33,10 +33,10 @@ public function test_is_new_date( $currentday_string, $previousday_string, $expe
-			array( '21.05.19', '20.05.19', 1 ),
-			array( '21.05.19', false, 1 ),
-		);
+		   return array(
```
```suggestion
		return array(
```

dhruvang21 commented on tests/phpunit/tests/functions/isNewDay.php:
```diff
@@ -33,10 +33,10 @@ public function test_is_new_date( $currentday_string, $previousday_string, $expe
-			array( '21.05.19', false, 1 ),
-		);
+		   return array(
+			   array( '21.05.19', '21.05.19', false ),
```
```suggestion
			array( '21.05.19', '21.05.19', false ),
```

dhruvang21 commented on tests/phpunit/tests/functions/isNewDay.php:
```diff
@@ -33,10 +33,10 @@ public function test_is_new_date( $currentday_string, $previousday_string, $expe
-		);
+		   return array(
+			   array( '21.05.19', '21.05.19', false ),
+			   array( '21.05.19', '20.05.19', true ),
```
```suggestion
			array( '21.05.19', '20.05.19', true ),
```

dhruvang21 commented on tests/phpunit/tests/functions/isNewDay.php:
```diff
@@ -33,10 +33,10 @@ public function test_is_new_date( $currentday_string, $previousday_string, $expe
+		   return array(
+			   array( '21.05.19', '21.05.19', false ),
+			   array( '21.05.19', '20.05.19', true ),
+			   array( '21.05.19', false, true ),
```
```suggestion
			array( '21.05.19', false, true ),
```

dhruvang21 commented on tests/phpunit/tests/functions/isNewDay.php:
```diff
@@ -33,10 +33,10 @@ public function test_is_new_date( $currentday_string, $previousday_string, $expe
+			   array( '21.05.19', '21.05.19', false ),
+			   array( '21.05.19', '20.05.19', true ),
+			   array( '21.05.19', false, true ),
+		   );
```
```suggestion
		);
```

prab18hat commented:
> * Updated @return type for is_new_day() from int to bool
> * Clarified return type in xmlrpc_getpostcategory()
> * Improved @param description for get_weekstartend()
> * Corrected grammar in build_query() docblock
> * Improved return description in get_status_header_desc()
> 
> Summary of Changes:
> 
> Standardized PHPDoc for is_new_day():
> 
> Updated the @return type from int to bool for clarity and modern PHPDoc standards.
> 
> Grammar and Docblock Improvements in functions.php:
> 
> xmlrpc_getpostcategory(): Clarified the return type and simplified awkward phrasing for readability.
> 
> get_weekstartend(): Improved the @param description for $start_of_week to clearly state “Default is an empty string.”
> 
> build_query(): Corrected grammar by adding the missing article and removing an unnecessary comma.
> 
> Old: Builds URL query based on an associative and, or indexed array.
> 
> New: Builds a URL query based on an associative or indexed array.
> 
> get_status_header_desc(): Updated return description to: “Status description if found, or an empty string otherwise.” for better clarity.
> 
> Trac ticket: Trac ticket: https://core.trac.wordpress.org/ticket/63892
> 
> **This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

Hi, I noticed this PR was closed after the Trac ticket was fixed. However, some of the docblock and grammar improvements here are not yet in trunk. If possible, could a maintainer please reopen this PR so I can update it, or should I open a new PR instead? Thank you!
